lemur/roles/service.py

```diff
@@ -44,6 +47,8 @@ def set_third_party(role_id, third_party_status=False):
     role = get(role_id)
     role.third_party = third_party_status
     database.update(role)
+
+    log_service.audit_log("update_role", role.name, f"Updated third_party_status={third_party_status}")
```

Comment:
isn't this very verbose? basically all GG for each login per each user?

Reply:
This will be maximum once per role, so once per GG or email. `set_third_party` is called only if existing role is not marked as third-party.
Most GG will be auto created by lemur as third party ([during create](https://github.com/Netflix/lemur/blob/master/lemur/auth/views.py#L151-L159) this flag is set than setting separately). So I believe `set_third_party` is not used frequently.